### ISSUE TITLE
React-Native v2.0.2 Video Sharing Issue

### ISSUE THREAD

**Issue by jplandry908** [jitsi/jitsi-meet#14592]
After upgrading from @jitsi/react-native-sdk v1.0.3 to v2.0.2, I am now unable to share my device's camera. This is happening on both iOS and Android. I see the following similar messages on both devices after pressing the video button.

INFO  2024-04-02T21:26:10.603Z [modules/RTC/RTCUtils.js] Got media constraints:  {"video":{"height":{"ideal":720,"max":720,"min":180},"frameRate":{"max":20,"min":5},"facingMode":"user"},"audio":false}
LOG  2024-04-02T21:26:11.050Z [modules/RTC/RTCUtils.js] onUserMediaSuccess
DEBUG  2024-04-02T21:26:11.051Z [modules/RTC/JitsiLocalTrack.js] <_setStream>:  Setting new MSID: b90ec448-2d98-372b-6e38-a3e8e3486863 6E5CA4A1-486F-47E3-8C71-27E0EB4321A4 on LocalTrack[undefined,video]

**Comment by saghul** [comment]
Please share the full log when joining.

**Comment by jplandry908** [comment]
Hi @saghul. Here is a link to the logs, starting when I press the join button within my app. Please note, I modified a few of the values (like DOMAIN) for security reasons since this Jitsi server is self-hosted.

https://privatebin.net/?5b5c5b05ab4a0719#AanYVwfjKc7HU3N7vFoduipWqtGPeDCt1vnMivzr9vFh

A couple things to note:
- If I download jitsi back to v1.0.3, it works correctly without any issues
- I have 2 versions of this app, one running react-native v0.72.12 (production) and the other running v0.73.6. I know you mentioned in another post that you do not support v0.73 yet, so my production app is still on v0.72 but I attempted this release on both version. I'm holding on v0.72 until it is support or seems to be working well.
- I receive the same test results on both Android and iOS. The app connects to the conference without and issues. Audio works well, but when I try to turn on video is does nothing.
- I also attempted to publish this app via the app stores (i.e. Test Flight) but received the same results
- These logs are from a physical iOS running from xcode debug and captured using Flipper.

Please let me know if I can be of further assistance.

**Comment by saghul** [comment]
@Calinteodor Can you PTAL?

**Comment by jplandry908** [comment]
Hi @Calinteodor. I have the Flipper App installed on my Mac that I use to capture and filter logs, but this is not using Expo. I will download the sample project and give it a try.

**Comment by Calinteodor** [comment]
Got it. Please do. Thank you!

**Comment by jplandry908** [comment]
Hi @Calinteodor. I was able to successfully test the sample app, which gave me the idea to point my app to 'meet.jit.si'. After changing the server to 'meet.jit.si', it works correctly. So the issue I am experiencing must not be related to the mobile app itself, although presented itself once upgrading from v1.0.3 to v2.0.2.

I just looked and the server is two versions behind. I'm wondering if upgrading the server will resolve the issue.

~$ apt list -a jitsi-meet
Listing... Done
jitsi-meet/stable 2.0.9364-1 all [upgradable from: 2.0.9220-1]
jitsi-meet/stable 2.0.9258-1 all
jitsi-meet/stable,now 2.0.9220-1 all [installed,upgradable to: 2.0.9364-1]

After I update the server, I'll run some more tests and provide another update in case anyone else is experiencing something similar.

**Comment by saghul** [comment]
Can you share your server config.js?

**Comment by jplandry908** [comment]
Hi @saghul. After updating the server to the latest version, I was still able to reproduce this issue. However, I discovered the root cause.

I compared my config file to the default and discovered that the issue was being caused by the constraints. This did not impact v1.0.3 but is impacting v2.0.2. If I remove the constraints, it works as expected. (Note: These constraints have been there for quite some time, so I am not certain why they were added or have a reason to keep then in the config).

These are the constraints that produce the issue.

`constraints: {
	video: {
		height: {
			ideal: 720,
			max: 720,
			min: 180,
		},
		frameRate: {
			max: 20,
			min: 5,
		},
	},
},
`

I tried various values for the constraints, but it appears just having the 'constraints' json declared in the file causes the issue.

`
	constraints: {
	},`

Removing this from the config, it works as expected.
